Automattic/jetpack
# Lazy Images: Attempt to prevent printing before images are loaded

Fixes #9232

#### Changes proposed in this Pull Request:


There's not a whole lot we can do, as browsers don't allow us to
actually delay or prevent the attempted print.

What we can do is:

* Trigger the loading of all lazy images on the page on print. They'll
  probably not load in time for the print though.
* Attempt to present an `alert()` informing the user that images aren't
  finished loading, and that they'll want to cancel the print and try
  again.
* Attempt to replace the printed page content with a message that
  images aren't finished loading, which will hopefully be seen in a
  "print preview" and at worst will avoid wastng ink.

#### Jetpack product discussion


None

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Create a post with several images.
* Verify lazy loading still works.
* Reload, and possibly clear the browser's file cache. Do not trigger loading of images on the reloaded page.
* Attempt to print. Probably you'll get an alert about images not being loaded yet, and the print preview may show the same message rather than the post's content.
* Wait for images to finish loading, and try printing again. There should be no alert and the print should print the post as expected.
* Reload, scroll down to trigger lazy loading of all images, and try printing. Again, if all the images finished loading there should be no alert and normal print output.

Do that in all supported browsers. I've done Firefox 81 and Chromium 83 on Linux and it seems to work.

#### Proposed changelog entry for your changes:

* Perhaps "Lazy Images: Trigger image loading on print, and attempt to warn the user if the print may contain not-yet-loaded images."
